bessctl/conf/testing/module_tests/mlfqueue.py

```diff
@@ -0,0 +1,36 @@
+import sugar
+from module import *
+from port import *
+
+def basic_throughput_test():
+    def gen_packet_list(protocol, input_ip, output_ip, num_pkts):
+        packet_list = []
+        for i in range(num_pkts):
+            cur_pkt = gen_packet(protocol, input_ip, output_ip)
+            packet_list.append({'input_port': 0, 'input_packet': cur_pkt,
+                                    'output_port': 0, "output_packet": cur_pkt})
+        return packet_list
+
+    out = []
+    single_basic = MLFQueue(num_lvls=5, batch_size=5, init_load=10.0)
+    single_packet = gen_packet(scapy.TCP, '22.22.22.22', '22.22.22.22')
+    out.append([single_basic,  # test this module
+                               1, 1,  # it has one input port and one output port
+                               [{'input_port': 0,
+                                   'input_packet': single_packet,  # send test_packet in on input port 0
+                                    'output_port': 0,
+                                    'output_packet': single_packet}]])
+    batch_basic = MLFQueue(num_lvls=5, batch_size=5, init_load=10.0)
+    packet_list = []
+    packet_list = gen_packet_list(scapy.TCP, '22.22.22.22', '22.22.22.22', 2)
+    packet_list += gen_packet_list(scapy.TCP, '31.14.15.69', '31.14.15.69', 3)
+    out.append([batch_basic,  # test this module
+                               1, 1,  # it has one input port and one output port
+                               packet_list])
+    return out
+
+def crash_test():
+    return [MLFQueue(num_lvls=5, batch_size=32, init_load=10.0), 1, 1]
+
```

Comment:
Would be good to have a test that actually looks at the core functionality of your Queue -- these tests just make sure the packets aren't modified and that the thing doesn't crash. But what about a test that shows that the Queue is balancing load as expected? (You can do this with a custom test.)